src/cmap/test/integration.rs

```diff
@@ -137,3 +138,73 @@ async fn concurrent_connections() {
         .await
         .expect("disabling fail point should succeed");
 }
+
+#[cfg_attr(feature = "tokio-runtime", tokio::test(threaded_scheduler))]
```

Comment:
I wonder if it's possible to take this a step further, e.g. automatically acquire the lock exclusively when we create a failpoint. This would presumably require us not to acquire the lock manually in the tests to avoid deadlocks, but maybe we could get around that by _always_ acquring a read lock and then passing that into the constructor for Failpoint (which then drops it and acquires a write lock instead). There wouldn't be any way to get the read lock back in the test function after the Failpoint is dropped, but I'm guessing there isn't really anywherte we'd need that.

Reply:
In a lot of cases we need to check the server version before we enable failpoints, so we definitely need to acquire at least a read lock beforehand. Also, we may need to repeatedly enable failpoints over the duration of a test (e.g. in a  spec test), so I don't think we can use the one that upgrades the lock either. I think we'll just have to rely on our diligence until we come up with a macro for automating the lock acquisition or something.

Reply:
Fair enough